### ISSUE TITLE
SSH to Docker containers does not work

### ISSUE THREAD

**Issue by shreyashguptas** [getcursor/cursor#1707]
1) If you can, please include a screenshot of your problem
![Screenshot 2024-08-26 at 1 40 07 PM](https://github.com/user-attachments/assets/7a64c7b3-3459-418e-9361-bac770e7da1b)

2) Please include the name of your operating system
MacOS 15.1 Beta (24B5024e)

3) If you can, steps to reproduce are super helpful
I SSH into my Linux Debian server and then attaching Visual Studio Code to the Docker container using the Docker extension in the IDE throws me an error. Doing the same exact thing in Visual Studio Code works everytime.

**Comment by yuchenlichuck** [comment]
not thing

**Comment by markNZed** [comment]
How are you connecting via SSH ? Are you using the icon in the bottom left corner of Cursor ? I was able to SSH into a machine and attach to a docker container with Cursor.

**Comment by shreyashguptas** [comment]
Yes so I am able to SSH into a machine but attaching a docker container using the docker extension where it shows all the running containers. I right click on it and click 'Attach Visual Studio Code' (yes in Cursor it shows Visual Studio Code, dont think they have changed that text). When I do the same thing in stock VS Code it works no problem but in cursor it does not.
![Screenshot 2024-09-18 at 9 01 40 AM](https://github.com/user-attachments/assets/b46030c4-170f-4ad7-b718-899eaa59bd95)

**Comment by markNZed** [comment]
I am using the VS Code extension ms-vscode-remote.remote-containers to attach, the command is: `Dev Containers: Attach to Running Container...`

**Comment by shreyashguptas** [comment]
Dev containers also does not work. I get the same 404 error. 

![Screenshot 2024-09-18 at 9 29 14 AM](https://github.com/user-attachments/assets/4d8bc08f-6e23-46a0-842d-835e6dd0a918)

**Comment by markNZed** [comment]
I am no expert in this but given that it is working for me then I guess the problem is not Cursor. Can you access that file yourself from the server (not in the container) and from inside the container too ?

**Comment by shreyashguptas** [comment]
Yes I can use my terminal to access all files in the container and also use VS Code to access it with Docker extension, Dev Container Extension. But does not work in Cursor for some reason.

**Comment by markNZed** [comment]
Imean can you access that URL with the ...alpine.tar.gz URL ? From the remote machine and from inside the container ? You could use something like wget to try and fetch the file

**Comment by shreyashguptas** [comment]
I can but my main thing is working with Jupyter notebooks inside a container and I can only do that if I can attached an IDE to the container. But I appreciate your help on this tho @markNZed

**Comment by markNZed** [comment]
I doubt I am of much help. My setup is a linux client and linux server, maybe that is playing a part. Good luck!

**Comment by lukemelas** [comment]
Hi, can you try with the latest version @shreyashguptas? For context, I work all day within a devcontainer in a remote SSH instance (Ubuntu), and I'm connecting from a Mac.

**Comment by shreyashguptas** [comment]
I am not sure what is wrong with my Cursor. I am able to connect my Debain Server with Remote SSH but not my RaspberryPi now. Neither can I connect into my docker containers on my Debian Server.

I may reinstall Cursor and see if anything fixes and play around with the settings.

**Comment by stijn-zyphra** [comment]
Currently having exactly this problem when trying to attach a Docker Container from Cursor through remote SSH.
Dev Containers version is 0.327.0 which is >1 year old. @Cursor team could you update it to a more recent version?
@shreyashguptas Did you find a workaround for this? Could write a custom script to port from VSCode Server to Cursor Server but this is not ideal.

<img width="245" alt="Image" src="https://github.com/user-attachments/assets/eb3ae12f-ac58-4968-b8cd-58c48182c108" />

**Comment by shreyashguptas** [comment]
No I just used VS Code to run dev containers.

**Comment by stijn-zyphra** [comment]
It' working in 0.45+

**Comment by shreyashguptas** [comment]
Awesome! will check it out soon. Appreciate it
